src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplayVisitor_Minimal.cs

```diff
@@ -159,6 +159,20 @@ private void MinimallyQualify(INamedTypeSymbol symbol)
             AddNameAndTypeArgumentsOrParameters(symbol);
         }
 
+        protected bool InObjectInitializatioContext(INamedTypeSymbol symbol)
```

Comment:
Is the check for `SyntaxKind.NewKeyword` necessary? If not, then could the issue be fixed by simplifying `NameBoundSuccessfullyToSameSymbol`, rather than introducing a new method here, since `NameBoundSuccessfullyToSameSymbol` is already calling `LookupNamespacesAndTypes`:
```
        protected bool NameBoundSuccessfullyToSameSymbol(INamedTypeSymbol symbol)
        {
            if (!ShouldRestrictMinimallyQualifyLookupToNamespacesAndTypes())
            {
                ImmutableArray<ISymbol> normalSymbols = semanticModelOpt.LookupSymbols(positionOpt, name: symbol.Name);
                ISymbol normalSymbol = SingleSymbolWithArity(normalSymbols, symbol.Arity);

                if (normalSymbol == null)
                {
                    return false;
                }

                // Binding normally ended up with the right symbol.  We can definitely use the
                // simplified name.
                if (normalSymbol.Equals(symbol.OriginalDefinition))
                {
                    return true;
                }
            }

            // Binding normally failed.  We may be in a "Color Color" situation where 'Color'
            // will bind to the field, but we could still allow simplification here.
            ImmutableArray<ISymbol> typeOnlySymbols = semanticModelOpt.LookupNamespacesAndTypes(positionOpt, name: symbol.Name);
            ISymbol typeOnlySymbol = SingleSymbolWithArity(typeOnlySymbols, symbol.Arity);

            if (typeOnlySymbol == null)
            {
                return false;
            }

            return typeOnlySymbol.Equals(symbol.OriginalDefinition);
        }
```